src/js/view/view.js

```diff
@@ -396,7 +396,14 @@ function View(_api, _model) {
                     }
                 }
             },
-            doubleClick: () => _controls && api.setFullscreen()
+            doubleClick: () => _controls && api.setFullscreen(),
+            longPress: (evt) => {
+                // manually trigger the right click menu on iOS
+                // because context menu is not supported
+                if (OS.iOS && _controls && _controls.rightClickMenu) {
```

Comment:
Instead of checking for iOS here, you should only create a `UI` instance that implements longPress event handling on iOS. This currently implementation makes all browsers run a timeout on mousedown or touchstart for an event that will do nothing.

Reply:
I don't think you should be tagging this on to the `clickHandlerHelper`. This should have it's own UI instance so we can change the element, and only create and register it for environments that need it.

Reply:
You could also implement the longpress without UI outside of here on player element. This would be better for performance since we only need to register touchstart for iOS and no other listeners.